Makefile

```diff
@@ -5,15 +5,22 @@ DIRS=tmp src tmp/.ccache dist
 
 ICU_REPO=http://source.icu-project.org/repos/icu/trunk
 BINPATH=/src/bin/
+# the 'full' list
 DISTROS=$(shell cd dockerfiles;ls)
+# the 'small' list. The first one is primary.
 DISTROS_SMALL=ubuntu fedora
 SRC=src/icu
-REV:=$(shell (cd $(SRC) && (git describe --tags --exact-match 2>/dev/null || git rev-parse --short HEAD)) || echo 'unknown')
+REV:=$(shell (cd $(SRC) >/dev/null 2>/dev/null && (git describe --tags --exact-match 2>/dev/null || git rev-parse --short HEAD)) || echo 'unknown')
```

Comment:
What's the purpose of the first `>/dev/null` here, under what circumstances would `cd` ever output anything to `stdout`?

Reply:
not sure which circumstances. but it is certainly doing so.  might be sh vs bash

Reply:
I'd suggest that you remove the first `>/dev/null` and in case you later actually encounter a case where it's needed you can then add it with an explanation of the circumstances under which it's needed.

Reply:
It was needed on my mac, otherwise I get  `REV=/some/path 64.2`. I will test again.

Reply:
mystery solved: complete user error, https://superuser.com/a/90537/13787

Reply:
In that case, I think that `unset CDPATH`, as recommended there, seems like the most straightforward solution, which leaves no doubt as to what the purpose of it is.

Reply:
> In that case, I think that `unset CDPATH`, as recommended there, seems like the most straightforward solution

I'm inclined to remove `>/dev/null 2>/dev/null` everywhere and (possibly) just document to not set `CDPATH`. 

Huh, i found something! See https://bosker.wordpress.com/2012/02/12/bash-scripters-beware-of-the-cdpath/ (kind of a public service announcement):
> You can avoid it in some cases by using cd ./foo, which does not consult CDPATH. But this is not a panacea: it can’t easily be used with paths that might be absolute or relative, such as `dirname "$0"`, so I think unsetting CDPATH is still the best way to deal with it.

That actually implies that `Q=$(cd ./stuff; echo bar)` or `P=$(cd /path/to/stuff; echo baz)` might be best - using a non-ambiguous path starting with `/` or `./`.  

So setting:

```makefile
SRC=./src/icu
```

…at the top of this Makefile actually avoids the CDPATH issue.